src/os/freertos/include/AbstractTask.tpp

```diff
@@ -27,6 +27,11 @@ bool AbstractTask<stackSize>::start() {
     return false;
 }
 
+template <unsigned int stackSize>
+bool AbstractTask<stackSize>::isRunning() {
+    return m_taskRunning;
```

Comment:
You could check the task status from FreeRTOS instead of the flag, or just check that m_taskHandler is not null

```c++
retrun eTaskGetState(m_taskHandle) == eRunning;
```

Reply:
Wouldn't it be possible for the task to be blocked or suspended by the RTOS if the scheduler needs to handle a higher priority task? So it would still be in the while loop but not in the running state?

Reply:
True, you would have to check if the task is either running, ready, or blocked. In other words, not suspended. You can keep the bool flag since we never suspend our tasks